Model/Export/Product.php

```diff
@@ -14,6 +14,42 @@
 class Product extends MagentoProduct
 {
     /**
+     * Export process
+     *
+     * @return string
+     */
+    public function export()
+    {
+        //Execution time may be very long
+        set_time_limit(0);
+
+        $writer = $this->getWriter();
+        $page = 0;
+        while (true) {
```

Comment:
Would it not be better to paginate the collection and then while page < collectionPageCount?